packages/vite/src/node/server/middlewares/rewriteUrl.ts

```diff
@@ -20,7 +21,9 @@ export function spaFallbackMiddleware(
           if (fs.existsSync(path.join(root, rewritten))) {
             return rewritten
           } else {
-            return `/index.html`
+            if (spaFallback) {
+              return `/index.html`
+            }
           }
         }
       }
```

Comment:
Few lines below we have:

```
  return function viteSpaFallbackMiddleware(req, res, next) {
    return historySpaFallbackMiddleware(req, res, next)
  }
```

so technically we're still preserving the old function name (so someone can remove it). If `appType` is the blessed way forward, we could say that the middleware function names aren't part of the public API following semver? And change it too? 🤔

Reply:
Oh, I didn't see that the name was not yet changed.

Yes, I think we should rename it to `htmlFallbackMiddleware` in this PR, since everyone in v3 that doesn't want the HTML middleware is using `appType: custom` I think we could do this change in Vite 3.1.